linkcheck/parser/__init__.py

```diff
@@ -125,7 +130,9 @@ def find_links (url_data, callback, tags):
     handler.parser = parser
     # parse
     try:
-        parser.feed(url_data.get_content())
+        content = url_data.get_content()
+        with parse_mutex:
+            parser.feed(content)
         parser.flush()
```

Comment:
I may be missing previous context, but it's unclear to me why `parser.feed()` is not thread-safe, but `parser.flush()` is fine?

Are we using the same parser instance in multiple threads?  (It doesn't look like we do.)

Is the parser using global state shared between multiple parser instances?

Is it perhaps GNU bison that's giving us a thread-unsafe parser?

Perhaps we should prioritize #119 that replaces our homegrown C parser with BeatifulSoup?